packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2849,15 +2819,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             const maxOpSize = this.context.deltaManager.maxMessageSize;
 
             // If in TurnBased flush mode we will trigger a flush at the next turn break
-            if (this.flushMode === FlushMode.TurnBased && !this.needsFlush) {
+            if (this.currentlyBatching() && !this.needsFlush) {
                 opMetadataInternal = {
                     ...opMetadata,
                     batch: true,
                 };
                 this.needsFlush = true;
 
                 // Use Promise.resolve().then() to queue a microtask to detect the end of the turn and force a flush.
-                if (!this.flushTrigger) {
+                if (this.flushMode === FlushMode.TurnBased && !this.flushTrigger) {
+                    this.flushTrigger = true;
```

Comment:
is `this.flushTrigger` actually used? I've noticed it before and nobody actually sets it to true. Can we just remove it?